common/graph_db_client.py

```diff
@@ -87,72 +88,79 @@ def element_exists(self,element_type:str,element_id:int,conditions:str="")->bool
-                        ".property(list,'text_unit_ids',prop_text_unit_ids)"
-                    ),
-                    bindings={
-                        "prop_id": row.id,
-                        "prop_name": row.name,
-                        "prop_type": row.type,
-                        "prop_description": row.description,
-                        "prop_human_readable_id": row.human_readable_id,
-                        "prop_partition_key": "entities",
-                        "prop_description_embedding":json.dumps(row.description_embedding.tolist() if row.description_embedding is not None else []),
-                        "prop_graph_embedding":json.dumps(row.graph_embedding.tolist() if row.graph_embedding is not None else []),
-                        "prop_text_unit_ids":json.dumps(row.text_unit_ids.tolist() if row.text_unit_ids is not None else []),
-                    },
+        step_df=50
+        split_dataframes=[data[i:i+step_df] for i in range(0,len(data),step_df)]
+        for spilt_df in split_dataframes:
+            q="g"
+            query_bindings={}
+            iter_row=0
+            for row in spilt_df.itertuples():
+                if self.element_exists("g.V()",row.id):
+                    continue
+                q+=(
+                    ".addV('entity')"
+                    ".property('id', prop_id"+str(iter_row)+")"
+                    ".property('name', prop_name"+str(iter_row)+")"
+                    ".property('type', prop_type"+str(iter_row)+")"
+                    ".property('description', prop_description"+str(iter_row)+")"
+                    ".property('human_readable_id', prop_human_readable_id"+str(iter_row)+")"
+                    ".property('category', 'entities')"
                 )
-            time.sleep(5)
+                query_bindings.update({
+                    ("prop_id"+str(iter_row)): row.id,
+                    ("prop_name"+str(iter_row)): row.name,
+                    ("prop_type"+str(iter_row)): row.type,
+                    ("prop_description"+str(iter_row)): row.description,
+                    ("prop_human_readable_id"+str(iter_row)): row.human_readable_id,
+                })
+                iter_row+=1
+            self._client.submit(
+                message=q,
+                bindings=query_bindings
+            )
 
 
     def write_edges(self,data: pd.DataFrame)->None:
-        for row in data.itertuples():
-            if self.element_exists("g.E()",row.id):
-                continue
-            self._client.submit(
-                message=(
-                    "g.V().has('name',prop_source_id)"
+        step_df=100
+        split_dataframes=[data[i:i+step_df] for i in range(0,len(data),step_df)]
+        for spilt_df in split_dataframes:
+            q="g"
+            query_bindings={}
+            iter_row=0
+            for row in spilt_df.itertuples():
+                if self.element_exists("g.E()",row.id):
```

Comment:
Are relationship ids unique between index runs? I will double check that because I know entities weren't until @sirus-ms made that change to sha256 the name.